npc/cities/aldebaran.txt

```diff
@@ -1266,18 +1262,20 @@ function	script	F_Lottery	{
 		if(@Lotto == 4) getitem Red_Potion,250;//Items: Red_Potion,
 		if(@Lotto == 5) getitem Red_Potion,300;//Items: Red_Potion,
 		return;
+	}
 
 sF_Spin:
-	mes "[Lottery Machine]";
-	mes "Number of spins remaining: "+@input;
-	next;
-	mes "[Lottery Machine]";
-	mes "(rumble~rumble~rumble~)...";
-	next;
-	@temp = rand(10);
-	--@input;
-	if(@input <= 0) return;
-	goto sF_Spin;
+	while (true) {
+		mes "[Lottery Machine]";
+		mes "Number of spins remaining: "+@input;
+		next;
+		mes "[Lottery Machine]";
+		mes "(rumble~rumble~rumble~)...";
+		next;
+		@temp = rand(10);
+		--@input;
+		if(@input <= 0) return;
+	}
```

Comment:
how about like this?
```
while (@input > 0) {
	mes "[Lottery Machine]";
	mes "Number of spins remaining: "+@input;
	next;
	mes "[Lottery Machine]";
	mes "(rumble~rumble~rumble~)...";
	next;
	@temp = rand(10);
	--@input;
}
return;
```
instead of `while(true)` .... use `while(@input > 0)`